Cubical/Structures/Relational/Macro.agda

```diff
@@ -0,0 +1,87 @@
+{-
+
+Descriptor language for easily defining relational structures
+
+-}
+{-# OPTIONS --cubical --no-import-sorts --no-exact-split --safe #-}
+module Cubical.Structures.Relational.Macro where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.RelationalStructure
+
+open import Cubical.Structures.Relational.Constant
+open import Cubical.Structures.Relational.Join
+open import Cubical.Structures.Relational.Maybe
+open import Cubical.Structures.Relational.Parameterized
+open import Cubical.Structures.Relational.Pointed
+open import Cubical.Structures.Relational.UnaryOp
+
+data RelDesc (ℓ : Level) : Typeω where
+  -- constant structure: X ↦ A
+  constant : ∀ {ℓ'} → hSet ℓ' → RelDesc ℓ
+  -- pointed structure: X ↦ X
+  var : RelDesc ℓ
+  -- join of structures S,T : X ↦ (S X × T X)
+  _,_ : RelDesc ℓ  → RelDesc ℓ  → RelDesc ℓ
+  -- structure S parameterized by constant A : X ↦ (A → S X)
+  param : ∀ {ℓ'} → (A : Type ℓ') → RelDesc ℓ  → RelDesc ℓ
+  -- structure S parameterized by variable argument: X ↦ (X → S X)
+  recvar : RelDesc ℓ  → RelDesc ℓ
+  -- Maybe on a structure S: X ↦ Maybe (S X)
+  maybe : RelDesc ℓ → RelDesc ℓ
+  -- arbitrary structure with notion of structured isomorphism given by functorial action
```

Comment:
This comment should probably be removed..

Reply:
whoops